nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -186,8 +186,10 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             try (final Connection con = dbcpService.getConnection();
                  final Statement st = con.createStatement()) {
 
-                final Integer queryTimeout = context.getProperty(QUERY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
-                st.setQueryTimeout(queryTimeout); // timeout in seconds
+                if(dbAdapter.getSupportsStatementTimeout()) {
```

Comment:
This makes sense to me, but when we were talking about supporting Hive in ExecuteSQL, there were concerns about those drivers that don't support statement timeouts, which is one reason why there's a separate Hive NAR (and Select/PutHiveQL processors). This was before DatabaseAdapter came along, so perhaps now this can be alleviated as you've done here. Does this work when using a HiveConnectionPool? Or are you using a DBCPConnectionPool that points to Hive JARs?  It seems like if this works with a HiveConnectionPool (with your changes to JdbcCommon below), then the Select/PutHiveQL processors could be deprecated.